app/services/moderator/merge_user.rb

```diff
@@ -55,45 +58,77 @@ def update_social
     end
 
+    def merge_sponsorships
+      @delete_user.sponsorships.update_all(user_id: @keep_user.id)
+    end
+
     def merge_profile
-      if @delete_user.github_repos.any?
-        @delete_user.github_repos.update_all(user_id: @keep_user.id)
-        @keep_user.touch(:github_repos_updated_at)
-      end
-      if @delete_user.badge_achievements.any?
-        @delete_user.badge_achievements.update_all(user_id: @keep_user.id)
-        BadgeAchievement.counter_culture_fix_counts(where: { users: { id: @keep_user.id } })
-      end
+      @delete_user.github_repos.update_all(user_id: @keep_user.id)
+      @keep_user.touch(:github_repos_updated_at)
 
       @keep_user.update_columns(created_at: @delete_user.created_at) if @delete_user.created_at < @keep_user.created_at
     end
 
-    def merge_chat_mentions
-      any_memberships = @delete_user.chat_channel_memberships.any?
-      @delete_user.chat_channel_memberships.update_all(user_id: @keep_user.id) if any_memberships
-      @delete_user.mentions.update_all(user_id: @keep_user.id) if @delete_user.mentions.any?
+    def merge_badge_achievements
+      duplicate_badges = @delete_user.badge_achievements.pluck(:badge_id).intersection(
+        @keep_user.badge_achievements.pluck(:badge_id),
+      )
+      @delete_user.badge_achievements.where(badge_id: duplicate_badges).delete_all
+      @delete_user.badge_achievements.update_all(user_id: @keep_user.id)
+      BadgeAchievement.counter_culture_fix_counts(where: { users: { id: @keep_user.id } })
     end
 
-    def merge_follows
-      @delete_user.follows&.update_all(follower_id: @keep_user.id) if @delete_user.follows.any?
-      @delete_user_followers = Follow.followable_user(@delete_user.id)
-      @delete_user_followers.update_all(followable_id: @keep_user.id) if @delete_user_followers.any?
+    def merge_chat_channels
+      duplicate_moderator_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active", role: "mod",
+      ).pluck(:chat_channel_id).intersection(
+        @keep_user.chat_channel_memberships.where(
+          status: "active", role: "mod",
+        ).pluck(:chat_channel_id),
+      )
+      @delete_user.chat_channel_memberships.where(chat_channel_id: duplicate_moderator_memberships).delete_all
+
+      moderator_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active", role: "mod",
+      ).pluck(:chat_channel_id)
+
+      duplicate_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active",
+      ).pluck(:chat_channel_id).intersection(
+        @keep_user.chat_channel_memberships.where(status: "active").pluck(:chat_channel_id),
+      )
+      @delete_user.chat_channel_memberships.where(chat_channel_id: duplicate_memberships).delete_all
+      @keep_user.chat_channel_memberships.where(chat_channel_id: moderator_memberships).update_all(role: "mod")
+      @delete_user.chat_channel_memberships.update_all(user_id: @keep_user.id)
     end
```

Comment:
Okay, this is the ridiculous part. Here, we have to account for both users' `status` and `role`. Because there are now two scopes to consider, this query is overly complicated and I think needs a refactor. I'll revisit this code, but wanted to get eyes on the code pattern first.